source/slang/slang-check-decl.cpp

```diff
@@ -10620,6 +10620,37 @@ bool getExtensionTargetDeclList(
 
 void SemanticsDeclBasesVisitor::_validateExtensionDeclTargetType(ExtensionDecl* decl)
 {
+    // Check if any generic parameter on the extension is not referenced by the target type
+    if (auto genericDecl = as<GenericDecl>(decl->parentDecl))
+    {
+        // Collect all generic parameters from the extension's generic declaration
+        List<Decl*> genericParams;
+        for (auto member : genericDecl->getDirectMemberDecls())
+        {
+            if (as<GenericTypeParamDecl>(member) || as<GenericValueParamDecl>(member))
```

Comment:
This should be `GenericTypeParamDeclBase`, also you don't need the `genericParams` list, just run the validation check directly in this loop.